tests/backends/test_docker.py

```diff
@@ -54,3 +56,73 @@ def test_custom_master_mounts(
             local_file.write(new_content)
             result = master.run_as_root(args=args)
             assert result.stdout.decode() == new_content
+
+
+class TestBadParameters:
+    """
+    Tests for bad parameters passed to Docker clusters.
+    """
+
+    def test_no_build_artifact(self, tmpdir: local) -> None:
+        """
+        The docker backend requires a build artifact in order
+        to launch a DC/OS cluster.
+        """
+        with pytest.raises(ValueError) as excinfo:
+            with Cluster(
+                cluster_backend=Docker(workspace_dir=tmpdir),
+                build_artifact=None,
+                masters=1,
+                agents=0,
+                public_agents=0,
+            ):
+                pass  # pragma: no cover
+
+        expected_error = (
+            'The Docker backend only supports creating new clusters. '
+            'Therefore any cluster created with the backend must '
+            'receive a build artifact.'
+        )
+
+        assert str(excinfo.value) == expected_error
+
+    def test_unsupported_build_artifact(
+        self, tmpdir: local, oss_artifact_url: str
+    ) -> None:
+        """
+        If the given build artifact is not a Path
+        the Docker backend will not accept it.
+        """
+        with pytest.raises(NotImplementedError) as excinfo:
+            with Cluster(
+                cluster_backend=Docker(workspace_dir=tmpdir),
+                build_artifact=oss_artifact_url,
+                masters=1,
+                agents=0,
+                public_agents=0,
+            ):
+                pass  # pragma: no cover
+
+        expected_error = (
+            'The Docker backend only supports creating clusters from '
+            'build artifacts specified by Path.'
+        )
+
+        assert str(excinfo.value) == expected_error
+
+    def test_invalid_build_artifact(self, tmpdir: local) -> None:
+        """
+        If the given Path does not point to a valid build artifact
+        the subprocess for calling DC/OS Docker will fail.
+        """
+        invalid_build_artifact = Path('/{}'.format(uuid.uuid4()))
```

Comment:
The Mesosphere style guide suggests that we use named format arguments. E.g. `'/{directory_name}'.format(directory_name=uuid.uuid4()`.